client/src/app/components/ContentListComponent.tsx

```diff
@@ -35,15 +40,54 @@ export default class ContentList extends React.Component<{
     await this.updateList();
   }
 
+  get filteredItems() {
+    const filters = this.state.search
+      .trim()
+      .toLocaleLowerCase()
+      .split(' ')
+      .map((s) => {
+        switch (true) {
+          case s.startsWith('id:'):
+            return (content: IContentListEntry) =>
+              content.contentId === s.slice(3);
+          case s.startsWith('lib:'):
+            return (content: IContentListEntry) =>
+              content.mainLibrary.toLocaleLowerCase().includes(s.slice(4));
+          default:
+            return (content: IContentListEntry) =>
+              content.title.toLocaleLowerCase().includes(s);
```

Comment:
Got an error in my local when the folder title is a number. So had to add a cast like this. 
`String(content.title).toLocaleLowerCase().includes(s);`

Reply:
I think this may be a bug in the underlying h5p libraries. When you demonstrated the problem, we saw that the title was a number. In the underlying h5p library, `title` is [defined](https://github.com/Lumieducation/H5P-Nodejs-library/blob/10ff6134812f896b70b526685f52f42131d7fbbc/packages/h5p-server/src/types.ts#L118) as being a string. When the JSON is [loaded](https://github.com/Lumieducation/H5P-Nodejs-library/blob/10ff6134812f896b70b526685f52f42131d7fbbc/packages/h5p-server/src/implementation/fs/FileContentStorage.ts#L335), however, there is no type conversion or type checking.

I can override this in our file storage implementation and add type conversion and checking so that the types will match what is defined.

Reply:
In the process of fixing this, I might have found another bug in the h5p libraries. In `FileContentStorage.listContent`, each directory in the content directory is checked to ensure it has a `h5p.json` file in it. These checks are all done in parallel. On systems that have more strict limitations on the number of file descriptors that can be opened at once, this will cause an error when the number of directories checked exceeds the file descriptor limit.

Since we all use MacOS/Linux/Gitpod, this should not be a problem because these systems default to unlimited file descriptors. I think it would only be a problem on Windows. Consequently, it is probably not worth fixing now. I just wanted to add a note here incase it becomes a problem in the future so it might be easier to track down.

Reply:
@TylerZeroMaster Please can you create a card that we can use to track this issue if it occurs in the future?

Reply:
@TylerZeroMaster I am in for adding the type conversion and checks in our file storage implementation.